src/select/__tests__/select.test.tsx

```diff
@@ -305,6 +305,28 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
     });
   });
 
+  test('shows filtering result text after typing', () => {
```

Comment:
Would be nice to have a few more tests covering how the filtering results text "competes" with the other types of text that can be displayed in the dropdown. Maybe this is something for the `dropdown-status` tests.